rj_gameplay/rj_gameplay/basic_play_selector.py

```diff
@@ -1,89 +1,271 @@
+        #  for now will be commented out
+        """if game_info.state == stp.rc.GameState.STOP:
+            return situations.Stop()
+        elif game_info.is_restart():
+            return self.__analyze_restart(world_state, heuristics)"""
         if world_state.game_info is None and self.curr_play is None:
-            return (self.curr_situation, basic_defense.BasicDefense())
+            return (situations.BasicDefense, basic_defense.BasicDefense())
+        elif (
+            heuristics.ball_pos == BallPos.OUR_BALL
+            or heuristics.ball_pos == BallPos.FREE_BALL
+        ):
+            return (situations.BasicOffense, basic_offense.BasicOffense())
+        elif heuristics.ball_pos == BallPos.THEIR_BALL:
+            return (situations.BasicDefense, basic_defense.BasicDefense())
+
+    @staticmethod
+    def __analyze_restart(
+        world_state: stp.rc.WorldState,
+        heuristics: HeuristicInformation,
+    ) -> stp.situation.ISituation:
+        game_info = world_state.game_info
+        if game_info.is_kickoff():
+            if game_info.our_restart:
+                if game_info.is_setup():
+                    return situations.PrepareKickoff()
+                elif game_info.is_ready():
+                    return situations.Kickoff()
+            else:
+                return situations.DefendKickoff()
+
+        elif game_info.is_penalty():
+            if game_info.our_restart:
+                if game_info.is_setup():
+                    return situations.PreparePenalty()
+                else:
+                    return situations.Penalty()
+            else:
+                if game_info.is_setup() or game_info.is_ready():
+                    return situations.PrepareDefendPenalty()
+                else:
+                    return situations.DefendPenalty()
+
+        elif game_info.is_direct():
+            if heuristics.field_loc == FieldLoc.ATTACK_SIDE:
+                if game_info.our_restart:
+                    return situations.OffensiveKickDirect()
+                else:
+                    return situations.DefendRestartOffensiveDirect()
+            elif heuristics.field_loc == FieldLoc.MIDFIELD:
+                if game_info.our_restart:
+                    return situations.MidfieldKickDirect()
+                else:
+                    return situations.DefendRestartMidfieldDirect()
+            elif heuristics.field_loc == FieldLoc.DEFEND_SIDE:
+                if game_info.our_restart:
+                    return situations.Restart()
+            else:
+                raise RuntimeError("Unknown field_loc {}".format(heuristics.field_loc))
 
-        if world_state.game_info is not None:
-            self.curr_situation = self.analyzer.analyze_situation(world_state)
-            for sit, play in PLAY_DICT.items():
-                if isinstance(self.curr_situation, sit):
-                    self.curr_play = play()
+        elif game_info.is_indirect():
```

Comment:
We can probably get rid of indirect situation, since all free kicks are direct now.